### PR TITLE
[AIRFLOW-7117] Honor self.schema in sql_to_gcs as schema to upload

### PR DESCRIPTION
**Pull request by whynick1**
---
Issue link: [AIRFLOW-7117](https://issues.apache.org/jira/browse/AIRFLOW-7117)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/operators/sql_to_gcs.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 from tempfile import NamedTemporaryFile
 
 import unicodecsv as csv
+from six import string_types
```
Comment (mik-laj) [inline airflow/providers/google/cloud/operators/sql_to_gcs.py]: We don't use six on master branch anymore.

File: airflow/providers/google/cloud/operators/sql_to_gcs.py [hunk 2]
```diff
@@ -251,18 +252,20 @@ def _get_col_type_dict(self):
...
         """
-        schema = [self.field_to_bigquery(field) for field in cursor.description]
+        schema = self.schema or [self.field_to_bigquery(field) for field in cursor.description]
 
         self.log.info('Using schema for %s', self.schema_filename)
         self.log.debug("Current schema: %s", schema)
         tmp_schema_file_handle = NamedTemporaryFile(delete=True)
-        tmp_schema_file_handle.write(json.dumps(schema, sort_keys=True).encode('utf-8'))
+        schema_json = json.dumps(schema, sort_keys=True) if isinstance(schema, list) else schema
+        tmp_schema_file_handle.write(schema_json.encode('utf-8'))
```
Comment (mik-laj) [inline airflow/providers/google/cloud/operators/sql_to_gcs.py]: It looks overly complicated. What do you think about the following code?
```python
        if self.shcema:
            schema_json = self.schema
            self.log.info("Using user schema")
        else:
            self.log.info("Starts generating schema")
            schema = self.schema or [self.field_to_bigquery(field) for field in cursor.description]            
            tmp_schema_file_handle = NamedTemporaryFile(delete=True)
            schema_json = json.dumps(schema, sort_keys=True)

        self.log.info('Using schema for %s', self.schema_filename)
        self.log.debug("Current schema: %s", schema)

        tmp_schema_file_handle.write(schema_json.encode('utf-8'))
```

File: airflow/providers/google/cloud/operators/sql_to_gcs.py (hunk 1 above)
Comment (whynick1) [reply airflow/providers/google/cloud/operators/sql_to_gcs.py]: Woops, I will remove that! Good catch.

File: airflow/providers/google/cloud/operators/sql_to_gcs.py (hunk 2 above)
Comment (whynick1) [reply airflow/providers/google/cloud/operators/sql_to_gcs.py]: Thanks for the review! I like the structure.
Just to be clear, `self.schema` could be either `str` or `list`, and we don't want to call `json.dumps()` on a string. How about this edited version? @mik-laj 
```
        if self.shcema:
            self.log.info("Using user schema")
            schema = self.schema
        else:
            self.log.info("Starts generating schema")
            schema = [self.field_to_bigquery(field) for field in cursor.description]            
        
        if isinstance(schema, list):
            schema = json.dumps(schema, sort_keys=True)

        self.log.info('Using schema for %s', self.schema_filename)
        self.log.debug("Current schema: %s", schema)

        tmp_schema_file_handle = NamedTemporaryFile(delete=True)
        tmp_schema_file_handle.write(schema.encode('utf-8'))
```

Comment (mik-laj) [reply airflow/providers/google/cloud/operators/sql_to_gcs.py]: LGTM

**Pull request merged**